CromIAM/src/main/scala/cromiam/cromwell/CromwellClient.scala

```diff
@@ -11,40 +11,51 @@ import akka.stream.ActorMaterializer
 import cromiam.cromwell.CromwellClient._
+import cromiam.instrumentation.CromIamInstrumentation
 import cromwell.api.{CromwellClient => CromwellApiClient}
 import cromiam.server.status.StatusCheckedSubsystem
 import cromwell.api.model.{WorkflowId, WorkflowLabels, WorkflowMetadata}
 import spray.json._
-
+import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContextExecutor, Future}
 
 /**
   * Provides a CromIAM specific handle for Cromwell communication
   *
   * FIXME: Look for ways to synch this up with the mothership
   */
-class CromwellClient(scheme: String, interface: String, port: Int, log: LoggingAdapter)(implicit system: ActorSystem,
+class CromwellClient(scheme: String, interface: String, port: Int, log: LoggingAdapter, serviceRegistryActorRef: ActorRef)(implicit system: ActorSystem,
                                                                                         ece: ExecutionContextExecutor,
                                                                                         materializer: ActorMaterializer)
-  extends SprayJsonSupport with DefaultJsonProtocol with StatusCheckedSubsystem {
+  extends SprayJsonSupport with DefaultJsonProtocol with StatusCheckedSubsystem with CromIamInstrumentation{
 
   val cromwellUrl = new URL(s"$scheme://$interface:$port")
   val cromwellApiVersion = "v1"
 
   override val statusUri = uri"$cromwellUrl/engine/$cromwellApiVersion/status"
 
+  override val serviceRegistryActor: ActorRef = serviceRegistryActorRef
+
   val cromwellApiClient: CromwellApiClient = new CromwellApiClient(cromwellUrl, cromwellApiVersion)
 
-  def collectionForWorkflow(workflowId: String, user: User): Future[Collection] = {
+  def collectionForWorkflow(workflowId: String, user: User, cromIamRequest: HttpRequest): Future[Collection] = {
     import CromwellClient.EnhancedWorkflowLabels
 
     log.info("Requesting collection for " + workflowId + " for user " + user.userId + " from metadata")
 
+    val startTimestamp = System.currentTimeMillis
+
     // Look up in Cromwell what the collection is for this workflow. If it doesn't exist, fail the Future
     cromwellApiClient.labels(WorkflowId.fromString(workflowId), headers = List(user.authorization)) flatMap {
       _.caasCollection match {
-        case Some(c) => Future.successful(c)
-        case None => Future.failed(new IllegalArgumentException(s"Workflow $workflowId has no associated collection"))
+        case Some(c) => {
+          sendTimingApi(makePathFromRequestAndResponseString(cromIamRequest, "success"), (System.currentTimeMillis - startTimestamp).millis, wfCollectionPrefix)
+          Future.successful(c)
+        }
+        case None => {
+          sendTimingApi(makePathFromRequestAndResponseString(cromIamRequest, "failed"), (System.currentTimeMillis - startTimestamp).millis, wfCollectionPrefix)
+          Future.failed(new IllegalArgumentException(s"Workflow $workflowId has no associated collection"))
+        }
```

Comment:
TOL possible DRYing

```    cromwellApiClient.labels(WorkflowId.fromString(workflowId), headers = List(user.authorization)) flatMap {
      _.caasCollection match {
        case Some(c) =>
          withTiming("success") {
            Future.successful(c)
          }
        case None =>
          withTiming("failed") {
            Future.failed(new IllegalArgumentException(s"Workflow $workflowId has no associated collection"))
          }
      }

    def withTiming[A](response: String)(op: => Future[A]): Future[A] = {
      sendTimingApi(makePathFromRequestAndResponseString(cromIamRequest, response), (System.currentTimeMillis - startTimestamp).millis, wfCollectionPrefix)
      op
    }
```

Reply:
Pedantry: should probably be "succeeded"/"failed" or "success"/"failure"

Reply:
Good point. Changing failed -> failure